src/data/set/pointwise/basic.lean

```diff
@@ -1222,6 +1222,15 @@ begin
     exact ⟨b, mem_inter (mem_smul_set.mpr ⟨a, ha, by simp [← this]⟩) hb⟩, },
 end
 
+@[simp, to_additive] lemma Union_inv_smul :
+  (⋃ (g : α), g⁻¹ • s) = (⋃ (g : α), g • s) :=
+(equiv.inv α).supr_congr $ λ g, rfl
+
+@[to_additive]
+lemma Union_smul_eq_set_of_exists {s : set β} :
+  (⋃ (g : α), g • s) = {a | ∃ (g : α), g • a ∈ s} :=
+by { ext a, simp_rw [← Union_inv_smul, mem_Union, ← mem_inv_smul_set_iff, mem_set_of_eq], }
```

Comment:
Here's a proof that makes explicit what the orthogonal steps @YaelDillies is talking about are:
```suggestion
by simp_rw [←Union_inv_smul, ←preimage_smul, set.preimage, ←Union_set_of]
```

Reply:
Thanks, I agree this is a better proof.